### ISSUE TITLE
FLAME parameters from Metrical Tracker

### ISSUE THREAD

**Issue by vipinsdk** [ShenhanQian/GaussianAvatars#67]
Hello Shenhan, 

Previously we discussed on the VHAP code repository regarding FLAME parameters for 6 cameras with different intrinsics and extrinsics and I was facing the issue of fitting the FLAME parameters as the keypoints were not not aligning properly. so, I decided to use the [Metrical Tracker](https://github.com/Zielon/metrical-tracker) code to generate the FLAME parameters with my camera parameters and  I have generated them successfully and to match the input to the Gaussian Avatars code, I generated the export_dataset similar to VHAP but when i train the model the code crashes because the xyz loss is `nan` .
I suspect the following reasons for the failure:

1.  The Metrical Tracker uses FLAME2020 and Gaussian Avatars uses 2023, I think the mesh shape could be different. Does this matter?
2. In my exported dataset the Transformation matrix per FRAME is in original camera space i.e., camera extrinsics from camera calibration, what should be the right transformation matrix?
3. The input images and Fg_masks in my case are of full human body and not masked as VHAP output, could this be the cause for the failure?
4. The FLAME parameters that i have generated are from the original video resolution that is 1920 x 1080, Should this matter or not?
5. Should the FLAME translation parameter be changed to place the mesh at the origin, my FLAME translation is not at origin i guess. 

I want to train the Gaussian Avatars from the generated FLAME parameters, could you please let me know the changes that needs to be incorporated and also the possible changes to hyperparameters. 

Thanks in advance. 

With Regards,
Vippin

**Comment by ShenhanQian** [comment]
Hi Vippin,

1. You need to load FLAME 2020 for GaussianAvatars (GA) if your tracked FLAME parameters are for version 2020. They are not compatible with version 2023.
2. I don't have enough information to comment on this question.
3. It can be a problem if you have the whole body in the images but still use FLAME, which is only for the head and prevents Gaussian splats from moving too far away. Meanwhile, the Gaussian splats have to float far away to reconstruct the whole body in the image. This conflict can lead to unstable optimization.
4. I don't see high resolution as a problem.
5. The FLAME can translate to anywhere as long as being consistent with the camera poses.

**Comment by vipinsdk** [comment]
Hello Shenhan, 

Thanks for the reply. In the point 2, I wanted to know that when we generate the NeRF style dataset with transforms.json, what the transformation matrix should contain for each frame, should it be World view transformation or it can be camera transformation of RT of shape [1,4,4]. 

Thanks in advance.

With Regards,
Vippin

**Comment by ShenhanQian** [comment]
# C**amera conventions**

- NeRSemble raw data: OpenCV, world2camera
- Tracker:
    - dataloader: OpenCV, world2camera → OpenGL, world2camera
    - **export: OpenGL, camera2world**
- Gaussian-Splatting
    - dataloader: OpenGL, camera2world → Colmap/OpenCV
        - camera_info
            - R: world2cam.T
            - T: world2cam
        - camera
            - world_view_transform: world2cam.T
            - ProjectionMatrix: K.T
            - full_proj_transform: world2cam.T @ K.T

**Comment by vipinsdk** [comment]
Hello Shenhan,

Thanks for the reply. Now, the camera convention is clear to me. 

With Regards,
Vippin
